### ISSUE TITLE
Symbolic links don't work for .tool-versions file

### ISSUE THREAD

**Issue by brennanfee** [asdf-vm/asdf#329]
### Steps to reproduce

It seems that when using either the "asdf local" or "asdf global" command that if .tool-versions is a symbolic link sed overwrites the symbolic link with a file.

Given that, at least for the $HOME/.tool-versions variation of the file, it may be common for people to use symlinks this presents a problem.  (I use symlinks to manage my dotfiles like I'm sure a lot of people do.)

I've isolated the lines of code with the problem (lines 28, 29 of version_commands.sh).  However, the challenge is creating a cross-platform (Linux, MacOS) fix.

I'll keep working on it and if I can get a line that works on both systems I'll submit a PR.

### Expected behavior

When I use "asdf global" the symlink I created for $HOME/.tool-versions is followed and the underlying file is edited.

### Actual behavior

When I use "asdf global" the file is correctly edited but no longer represents a symlink to the original file, and the original file is not updated.

### Environment

asdf version: v0.5.0

**Comment by Stratus3D** [comment]
Good find!

**Comment by brennanfee** [comment]
@Stratus3d Thanks.  I wanted to briefly express to you and the rest working on ASDF just how much we appreciate it.  I had intended on making a go at this one this last weekend but, alas, my time and energy were not in alignment.  (Besides, it has been literally months since I have booted up my Mac - I'm a Linux guy nearly 100% these days).  Time permitting I may make an attempt at this during the week or over the next weekend.
